src/docs/introduction.rst

```diff
@@ -1,12 +1,102 @@
+create a user, update their password, remove a user. However, that’s
+pretty mean and a waste of time when there is a much easier way of
+doing things.
+
+The community project FreeIPA *(Free Identity Policy and Auditing)* is
+an Identity Management solution. It’s also known as Red Hat IdM, which
+is the downstream name. FreeIPA, Red Hat IdM uses 389DS
+*(Directory Server)* for its database.
+
+These directory servers use the LDAP *(Lightweight Directory Access Protocol)*
+network protocol to representing objects to the computers on the network.
+Objects like a user, then that user object contains his user information
+like given name, surname, phone number, etc.
+
+*Please do not confuse LDAP with 389DS, 389DS is a directory server that
+uses the LDAP protocol on port 389.*
+
+Microsoft *(Active Directory Domain Services)* ADDS, but abbreviated to
+Microsoft AD. AD is another identity management solution for Windows. It has
+an LDAP server and shares many components with IPA. IPA and AD can have
+a trust created between the domain domains. Why? Consider if there was a
+company merger or acquisition. There won’t be a need to migrate users but
+instead, set up a trust and allow Bob from RedHat to administer
+the IBM domain…
+
+It’s not uncommon for companies to have two departments, one that supports
+and administers Linux and the other team does Windows, so each will have
+control over its own users and groups. Using SSSD’s ipa_provider,  AD domain
+users can then log in to a host managed by IPA because of the established
+trust between domains.
+
+SSSD can authenticate directly to AD, using the ad_provider, and they’ll
+be no need for IPA servers. Users and groups will be entirely managed by
+AD. So why use indirect authentication? IPA is more flexible for managing
+Linux hosts than AD since it can manage SUDO rules, SELinux users and
+contexts, automount maps, and more. If all users are created in AD, but
+the company has two IT teams. The Linux team may have to wait for users
+to be created. Hopefully, you can start to see that each solution has its
+pros and cons and will not be discussed in this introduction.
+
+For more information about FreeIPA and other compatible directory servers,
+please check out the following links.
+
+- `FreeIPA <https://www.freeipa.org/page/Main_Page>`_
+- `389DS <https://directory.fedoraproject.org/>`_
+- `OpenLDAP <https://www.openldap.org/>`_
+- `Kerberos <https://web.mit.edu/kerberos/>`_
+- `Microsoft ADDS <https://www.microsoft.com/en-ie/windows-server>`_
+
+The ad_provider and ipa_provider are just some providers that can be used
+for authentication or authorization. Other providers is the ldap_provider,
+krb_provider, proxy_provider, and local_provider. For more information please
+check out the following MAN pages for more information.
+
+- sssd-ldap
+- sssd-krb5
+- sssd-ipa
+- sssd-ad
+- sssd-files
+
+What else is important to know about SSSD? We’ve discussed authentication
```

Comment:
I would add authentication vs authorization definition here just for clarity.

Reply:
What else is important to know about SSSD? We’ve discussed authentication,
verifying the user and that the correct password is being used. We have
not discussed if that user is **allowed** to login, this is what we mean by
authorization. 

?

Reply:
Sounds good